geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionClearMessageTest.java

```diff
@@ -0,0 +1,285 @@
+
+public class PartitionedRegionClearMessageTest {
+
+  private Collection<InternalDistributedMember> recipients;
+  private DistributionManager distributionManager;
+  private PartitionedRegion partitionedRegion;
+  private ReplyProcessor21 replyProcessor21;
+  private Object callbackArgument;
+  private EventID eventId;
+  private RegionEventFactory regionEventFactory;
+
+  @Before
+  public void setUp() {
+    recipients = emptySet();
+    distributionManager = mock(DistributionManager.class);
+    partitionedRegion = mock(PartitionedRegion.class);
+    replyProcessor21 = mock(ReplyProcessor21.class);
+    callbackArgument = new Object();
+    eventId = mock(EventID.class);
+    regionEventFactory = mock(RegionEventFactory.class);
+  }
+
+  @Test
+  public void construction_throwsNullPointerExceptionIfRecipientsIsNull() {
+    Throwable thrown = catchThrowable(() -> {
+      new PartitionedRegionClearMessage(null, distributionManager, 1,
+          replyProcessor21, OperationType.OP_PR_CLEAR, callbackArgument, eventId, false,
+          regionEventFactory);
+    });
+
+    assertThat(thrown).isInstanceOf(NullPointerException.class);
+  }
+
+  @Test
+  public void construction_findsAllDependencies() {
+    boolean isTransactionDistributed = true;
+    int regionId = 10;
+    InternalCache cache = mock(InternalCache.class);
+    RegionEventImpl regionEvent = mock(RegionEventImpl.class);
+    TXManagerImpl txManager = mock(TXManagerImpl.class);
+    when(cache.getTxManager()).thenReturn(txManager);
+    when(partitionedRegion.getCache()).thenReturn(cache);
+    when(partitionedRegion.getDistributionManager()).thenReturn(distributionManager);
+    when(partitionedRegion.getPRId()).thenReturn(regionId);
+    when(regionEvent.getEventId()).thenReturn(eventId);
+    when(regionEvent.getRawCallbackArgument()).thenReturn(callbackArgument);
+    when(txManager.isDistributed()).thenReturn(isTransactionDistributed);
+
+    PartitionedRegionClearMessage message = new PartitionedRegionClearMessage(recipients,
+        partitionedRegion,
+        replyProcessor21,
+        OperationType.OP_PR_CLEAR,
+        regionEvent);
+
+    assertThat(message.getDistributionManagerForTesting()).isSameAs(distributionManager);
+    assertThat(message.getCallbackArgumentForTesting()).isSameAs(callbackArgument);
+    assertThat(message.getRegionId()).isEqualTo(regionId);
+    assertThat(message.getEventID()).isEqualTo(eventId);
+    assertThat(message.isTransactionDistributed()).isEqualTo(isTransactionDistributed);
+
+    RegionEventFactory regionEventFactory = message.getRegionEventFactoryForTesting();
+    RegionEvent<?, ?> created =
+        regionEventFactory.create(partitionedRegion, Operation.DESTROY, callbackArgument, false,
+            mock(DistributedMember.class), mock(EventID.class));
+    assertThat(created).isInstanceOf(RegionEventImpl.class);
+  }
+
+  @Test
+  public void construction_setsTransactionDistributed() {
```

Comment:
This test case seems to be covered in the test above it. Is it necessary to have a seperate test for it here too?